test/prototype_common_utils.py

```diff
@@ -34,26 +133,104 @@ def __iter__(self):
         yield self.args
         yield self.kwargs
 
-    def __str__(self):
-        def short_repr(obj, max=20):
-            repr_ = repr(obj)
-            if len(repr_) <= max:
-                return repr_
+    def load(self, device="cpu"):
+        args = tuple(arg.load(device) if isinstance(arg, TensorLoader) else arg for arg in self.args)
+        kwargs = {
+            keyword: arg.load(device) if isinstance(arg, TensorLoader) else arg for keyword, arg in self.kwargs.items()
+        }
+        return args, kwargs
 
-            return f"{repr_[:max//2]}...{repr_[-(max//2-3):]}"
+    def __repr__(self):
+        def better_repr(obj):
+            if isinstance(obj, enum.Enum):
+                return str(obj)
+            else:
+                return repr(obj)
 
         return ", ".join(
             itertools.chain(
-                [short_repr(arg) for arg in self.args],
-                [f"{param}={short_repr(kwarg)}" for param, kwarg in self.kwargs.items()],
+                [better_repr(arg) for arg in self.args],
+                [f"{param}={better_repr(kwarg)}" for param, kwarg in self.kwargs.items()],
             )
         )
 
 
-make_tensor = functools.partial(torch.testing.make_tensor, device="cpu")
+DEFAULT_SQUARE_IMAGE_SIZE = (16, 16)
+DEFAULT_LANDSCAPE_IMAGE_SIZE = (7, 33)
+DEFAULT_PORTRAIT_IMAGE_SIZE = (31, 9)
+DEFAULT_IMAGE_SIZES = (DEFAULT_LANDSCAPE_IMAGE_SIZE, DEFAULT_PORTRAIT_IMAGE_SIZE, DEFAULT_SQUARE_IMAGE_SIZE)
+
+DEFAULT_EXTRA_DIMS = ((), (0,), (4,), (2, 3), (5, 0), (0, 5))
+
+
+def from_loader(loader_fn):
+    def wrapper(*args, **kwargs):
+        loader = loader_fn(*args, **kwargs)
+        return loader.load(kwargs.get("device", "cpu"))
+
+    return wrapper
+
+
+def from_loaders(loaders_fn):
+    def wrapper(*args, **kwargs):
+        loaders = loaders_fn(*args, **kwargs)
+        for loader in loaders:
+            yield loader.load(kwargs.get("device", "cpu"))
+
+    return wrapper
+
+
+class TensorLoader:
+    def __init__(self, fn, *, shape, dtype):
+        self.fn = fn
+        self.shape = shape
+        self.dtype = dtype
+
+    def unwrap(self):
+        return TensorLoader(
+            lambda shape, dtype, device: torch.Tensor(self.fn(shape, dtype, device)), shape=self.shape, dtype=self.dtype
+        )
```

Comment:
This is useful in case we have a loader for a subclass, e.g. `ImageLoader`, but want to test a kernel that needs to work on plain tensors. You will see a lot of these `image_loader.unwrap()` calls in the sample inputs functions in the kernel tests.

Reply:
I've removed this once again, since tests like `test_batched_vs_single` need this information before the kernel is called. If a kernel actually depends on the subclass, the `test_scripted_vs_eager` will catch it, since subclass won't get scripted.